test/xspattern.tests.ts

```diff
@@ -197,6 +233,17 @@ describe('xspattern', () => {
 		check('[abcd-[cdef]]', ['a', 'b'], ['c', 'd', 'e', 'f', 'g']);
 		check('[a-z-[aeoui]]', ['b', 'd', 'z'], ['a', 'u', 'o']);
 		check('[a--[a]]', ['-'], ['a', 'b']);
+		check('a[^b]', ['a ', 'ax', 'abaxb'], ['a', 'ab', 'abc'], true);
+	});
+
+	it('throws the correct error when it sees backreferences', () => {
```

Comment:
Could you also add a few tests that show that XPath-specific syntax such as backreferences throws appropriate parsing errors (or may be interpreted as a different type of expression) in XSD mode?